src/github_utils.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+import requests
+from github import Github
+
+def get_pull_request_diff():
+    """Fetch the pull request diff content from GitHub"""
+    g = Github(os.getenv('GIT_TOKEN'))
+
+    # Parse GitHub event context
+    event_path = os.getenv('GITHUB_EVENT_PATH')
+    with open(event_path, 'r') as f:
+        event_data = json.load(f)
+
+    repo = g.get_repo(event_data['repository']['full_name'])
+    pr_number = event_data['pull_request']['number']
+    pull_request = repo.get_pull(pr_number)
+
+    # Get diff content
+    diff_url = pull_request.diff_url
+    response = requests.get(diff_url)
+    return response.text
+
+
+def get_valid_lines(diff_content):
+    """Parse diff to find valid line numbers with proper initialization"""
+    valid_lines = {}
+    current_file = None
+    line_number = None
+    
+    for line in diff_content.split('\n'):
+        # Reset tracking for new files
+        if line.startswith('diff --git'):
+            current_file = line.split(' b/')[-1].strip()
+            valid_lines[current_file] = set()
+            line_number = None  # Reset for new file
+            continue
+            
+        # Parse hunk header
+        if line.startswith('@@'):
+            try:
+                # Extract new file line number (after +)
+                hunk_header = line.split('+', 1)[1].split(' ', 1)[0]
+                line_number = int(hunk_header.split(',')[0])
+                valid_lines[current_file].add(line_number)
+            except (IndexError, ValueError):
+                line_number = None  # Invalid hunk format
+            continue
+            
+        # Only process lines if we have valid line_number
+        if line_number is None:
+            continue
+            
+        # Track added lines
+        if line.startswith('+'):
+            valid_lines[current_file].add(line_number)
```

Comment:
The current logic adds the `line_number` to `valid_lines[current_file]` even within the hunk header parsing `try` block before confirming the number is correct.  If the `int()` fails, there will be incorrect line number.

```suggestion
Move `valid_lines[current_file].add(line_number)` into the `try` block after line number is confirmed to be a valid integer
```